packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart

```diff
@@ -575,26 +574,20 @@ class SignInStateMachine extends StateMachine<SignInEvent, SignInState> {
       ..refreshToken = refreshToken
       ..idToken = idTokenJwt;
 
-    dispatch(
+    await dispatch(
       CredentialStoreEvent.storeCredentials(
         CredentialStoreData(
           userPoolTokens: user.userPoolTokens.build(),
         ),
       ),
     );
 
-    // Clear anonymous credentials, if there were any, and fetch authenticated
+    // Upgrade anonymous credentials, if there were any, or fetch authenticated
     // credentials.
     if (hasIdentityPool) {
-      dispatch(
-        CredentialStoreEvent.clearCredentials(
-          CognitoIdentityPoolKeys(identityPoolConfig!),
-        ),
-      );
-
       await dispatch(
         const FetchAuthSessionEvent.fetch(
-          CognitoSessionOptions(getAWSCredentials: true),
+          CognitoSessionOptions(getAWSCredentials: true, forceRefresh: true),
```

Comment:
Wasn't this a separate PR?

Reply:
It was a separate PR to add the option. This PR included a commit to enhance the changeover from unauthenticated to authenticated. The cleanest way to do that changeover is to make use of the forceRefresh property.